### ISSUE TITLE
analysis fails on list comprehensions

### ISSUE THREAD

**Issue by timsgardner** [arcadia-unity/Arcadia#379]
```clojure
(require '[magic.analyzer :as mana])

(mana/analyze
    '(for [x []]
       :bla)
    (mana/empty-env))
```
throws:
```
clojure.lang.ExceptionInfo: Could not find instance method nth with args [System.Int32] for type clojure.lang.IChunk while analyzing form (.nth c__4498__auto__ i__53280)
* magic/analyzer/errors/evalfn
* clojure.lang.MultiFn
* magic/analyzer/analyze-host-forms/analyze-host-call
```

**Comment by timsgardner** [comment]
Does work if magic passes are excluded:
```
(defn other-analyze
  ([form] (other-analyze form (mana/empty-env)))
  ([form env] (other-analyze form env mana/run-passes))
  ([form env passes-fn]
   (binding [ana/macroexpand-1 mana/macroexpand-1
             ana/create-var    (fn [sym env]
                                 (doto (intern (:ns env) sym)
                                   (reset-meta! (meta sym))))
             ana/parse         mana/parse
             ana/var?          var?]
     (env/with-env (mana/global-env)
       (ana/analyze form env)))))

(other-analyze '(for [x []] :bla))
;; prints a humongous ast with no errors
```

**Comment by nasser** [comment]
This was an analysis bug, fixed in 2315cfa081d2d0223c5be2d2058b9ecd2ee81fe4. I am leaving this open as compiling that form fails to verify with

```
In method: user$fn$9438::invoke()
Not Verifiable: Invalid type Int32 at stack for unbox.any operation at 0x015c
Not Verifiable: Compare instruction applyed to ill formed stack (int (Int32) x object ([null] Complex)) at 0x014f
Not Verifiable: Expected a reference type on stack for isinst but found int (Int32) at 0x0144
```

**Comment by nasser** [comment]
On second tought, closing this in favor of nasser/magic#106
